Kernel/Process.cpp

```diff
@@ -533,21 +533,37 @@ bool Process::dump_perfcore()
     VERIFY(is_dumpable());
     VERIFY(m_perf_event_buffer);
     dbgln("Generating perfcore for pid: {}", pid().value());
-    auto description_or_error = VirtualFileSystem::the().open(String::formatted("perfcore.{}", pid().value()), O_CREAT | O_EXCL, 0400, current_directory(), UidAndGid { uid(), gid() });
-    if (description_or_error.is_error())
+    auto base_filename = String::formatted("perfcore.{}", pid().value());
+
+    auto description_or_error = VirtualFileSystem::the().open(base_filename, O_CREAT | O_EXCL, 0400, current_directory(), UidAndGid { uid(), gid() });
+
+    for (size_t attempt = 1; attempt < 10 && description_or_error.is_error(); ++attempt)
+        description_or_error = VirtualFileSystem::the().open(String::formatted("{}.{}", base_filename, attempt), O_CREAT | O_EXCL, 0400, current_directory(), UidAndGid { uid(), gid() });
+
+    if (description_or_error.is_error()) {
+        dbgln("Failed to generate perfcore for pid {}: Could not find non-existing filename.", pid().value());
         return false;
-    auto& description = description_or_error.value();
+    }
+
+    auto& description = *description_or_error.value();
     KBufferBuilder builder;
-    if (!m_perf_event_buffer->to_json(builder))
+    if (!m_perf_event_buffer->to_json(builder)) {
+        dbgln("Failed to generate perfcore for pid {}: Could not serialize performance events to JSON.", pid().value());
         return false;
+    }
 
     auto json = builder.build();
-    if (!json)
+    if (!json) {
+        dbgln("Failed to generate perfcore for pid {}: Could not not allocate buffer.", pid().value());
```

Comment:
Failing to not allocate a buffer is quite an interesting problem :P

Reply:
Yes, indeed.